content/blog/2023/11/2023-11-06-introducing-2-2-2-java-support-plan.adoc

```diff
@@ -0,0 +1,124 @@
+for only as long as it takes for the next Java LTS release to be delivered; i.e., for two years rather than six.
+This strategy would benefit developers and early adopters,
+and these benefits go beyond the mere availability of shiny new language features for developers —
+newer Java runtimes have implemented a significant number of performance optimizations that are of interest to operators,
+and an increasing number of third-party library dependencies require the adoption of a recent Java LTS release
+in order to receive bug fixes and security patches.
+
+Additionally, the Jenkins ecosystem consists of thousands of loosely-connected components,
+and building and testing them across all supported Java versions is a nontrivial effort,
+notwithstanding recent advances in dependency updating.
+Reducing the build and test matrix to a single version would save hundreds of thousands of dollars in cloud costs,
+to say nothing of the savings in development costs
+that are associated with the decreased cognitive load of a simpler build and test matrix.
+However, this strategy fails to meet the needs of large scale users,
+who want to adopt a Java LTS release and stick with it for as long as possible
+before upgrading to the next Java LTS release.
+
+image:/images/post-images/2023/11/06/introducing-2-2-2-java-support-plan-figure3.svg[Figure 3,title="Rejected proposal: Support one Java LTS release at any given time"]
+
+=== Accepted proposal: 2 + 2 + 2 Java support plan
+
+The above discussion highlights two categories of users,
+each of whose needs are legitimate but, through no fault of their own, are in conflict with each other.
+The natural solution, then, is to compromise at the midpoint.
+Therefore, the Jenkins project is adopting a 2 + 2 + 2 Java support plan, where Jenkins
+supports a new Java LTS release in the first two years after the general availability of that Java LTS release,
+then requires that Java LTS release as the Jenkins minimum Java version in the next two years of that Java LTS release’s upstream support,
+then drops support for that Java LTS release two years before that Java LTS release reaches end of life upstream.
+
+image:/images/post-images/2023/11/06/introducing-2-2-2-java-support-plan-figure4.svg[Figure 4,title="Accepted proposal: 2 + 2 + 2 Java support plan"]
+
+In practice, this means that Jenkins will support a given Java LTS release
+for approximately two-thirds the amount of time that upstream Java vendors do,
+that Jenkins will support two Java LTS releases at any given time rather than three,
+and that large scale users can stay on a Java LTS release for four years at a time.
+This plan balances the needs of large scale users for predictability and stability
+with the needs of early adopters and developers
+to improve and simplify Jenkins with the latest Java capabilities
+and to reduce the maintenance overhead associated with a large build and test matrix.
```

Comment:
```suggestion
This plan balances the needs of large scale users for predictability and stability,
with the needs of early adopters and developers,
to improve and simplify Jenkins with the latest Java capabilities
and to reduce the maintenance overhead associated with a large build and test matrix.
```
Small suggestion to add some punctuation for sentence structure/avoiding a run on sentence.

Reply:
A run-on sentence occurs when two independent clauses run together without proper punctuation or appropriate conjunctions. A clause requires both a subject and a verb. This sentence has only one subject: "plan". The only other nouns in this sentence are functioning as direct objects or objects of prepositions. So there is no run-on sentence here.

> This plan balances the needs of A for B with the needs of C to D and to E.

A = large scale users
B = predictability and stability
C = early adopters and developers
D = improve and simplify Jenkins with the latest Java capabilities
E = reduce the maintenance overhead associated with a large build and test matrix

Adopting your suggestion would result in the structure

> This plan balances the needs of A for B, with the needs of C, to D and to E.

This places commas before two prepositions, which is grammatically incorrect for the reasons given in https://thecriticalreader.com/no-comma-before-or-after-a-preposition/